core/src/execution/datafusion/expressions/scalar_funcs/chr.rs

```diff
@@ -44,10 +44,18 @@ pub fn chr(args: &[ArrayRef]) -> Result<ArrayRef> {
         .iter()
         .map(|integer: Option<i64>| {
             integer
-                .map(|integer| match core::char::from_u32(integer as u32) {
-                    Some(integer) => Ok(integer.to_string()),
-                    None => {
-                        exec_err!("requested character too large for encoding.")
+                .map(|integer| {
+                    let adjusted_integer = if integer >= 0 { integer % 256 } else { integer };
+
+                    match core::char::from_u32(adjusted_integer as u32) {
+                        Some(integer) => Ok(integer.to_string()),
+                        None => {
+                            if integer < 0 {
```

Comment:
nit: could you check for negative integers earlier, before calling `core::char::from_u32(adjusted_integer as u32)`. If the input is a negative number then we should not attempt to convert to a char.